### DISCUSSION TITLE
does flutter_rust_bridge::spawn need to be in async context

### DISCUSSION THREAD

**Discussion by jbbjarnason** [fzyzcjy/flutter_rust_bridge#2364] [Q&A] [answered]
I tried the following:
![image](https://github.com/user-attachments/assets/bf737b0c-affa-48e6-b2da-005c192cb121)

And got a runtime error `there is no reactor running, must be called from the context of a Tokio 1.x runtime`

Then I made the function async:
![image](https://github.com/user-attachments/assets/39d32064-e0e4-49cb-a354-09f2a26c8b20)

And it worked.

Is there a reason that this is like it is?

**Comment by fzyzcjy** [comment, marked as answer]
It seems to be a requirement of tokio instead of flutter_rust_bridge IIRC. e.g. google `tokio spawn there is no reactor running`
